src/pstore/pstore.c

```diff
@@ -0,0 +1,735 @@
+        _PSTORE_STORAGE_MAX,
+        _PSTORE_STORAGE_INVALID = -1
+} PstoreStorage;
+
+static const char* const pstore_storage_table[_PSTORE_STORAGE_MAX] = {
+        [PSTORE_STORAGE_NONE] = "none",
+        [PSTORE_STORAGE_ARCHIVE] = "archive",
+        [PSTORE_STORAGE_JOURNAL] = "journal",
+};
+
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP(pstore_storage, PstoreStorage);
+static DEFINE_CONFIG_PARSE_ENUM(config_parse_pstore_storage, pstore_storage, PstoreStorage, "Failed to parse storage setting");
+
+static PstoreStorage arg_storage = PSTORE_STORAGE_ARCHIVE;
+
+static bool arg_allowunlink = true;
+static char *arg_sourcedir = NULL;
+static char *arg_archivedir = NULL;
+STATIC_DESTRUCTOR_REGISTER(arg_sourcedir, freep);
+STATIC_DESTRUCTOR_REGISTER(arg_archivedir, freep);
+
+static int parse_config(void) {
+    int rc;
+    static const ConfigTableItem items[] = {
+            { "Pstore", "AllowUnlink",      config_parse_bool,              0, &arg_allowunlink },
+            { "Pstore", "Storage",          config_parse_pstore_storage,    0, &arg_storage     },
+            { "Pstore", "SourceDir",        config_parse_path,              0, &arg_sourcedir   },
+            { "Pstore", "ArchiveDir",       config_parse_path,              0, &arg_archivedir  },
+            {}
+    };
+
+    rc = config_parse_many_nulstr(PKGSYSCONFDIR "/pstore.conf",
+                                    CONF_PATHS_NULSTR("systemd/pstore.conf.d"),
+                                    "Pstore\0",
+                                    config_item_table_lookup, items,
+                                    CONFIG_PARSE_WARN, NULL);
+    if (NULL == arg_sourcedir)
+    {
+        arg_sourcedir = (char *)malloc(sizeof(ARG_SOURCEDIR_DEFAULT)+1);
+        if (NULL != arg_sourcedir)
+            strcpy(arg_sourcedir, ARG_SOURCEDIR_DEFAULT);
+    }
+    if (NULL == arg_archivedir)
+    {
+        arg_archivedir = (char *)malloc(sizeof(ARG_ARCHIVEDIR_DEFAULT)+1);
+        if (NULL != arg_archivedir)
+            strcpy(arg_archivedir, ARG_ARCHIVEDIR_DEFAULT);
+    }
+    return rc;
+}
+
+/*
+ * File list handling
+ */
+typedef struct dirent_cs
+{
+    struct dirent de;
+    struct stat statbuf;
+    int is_binary;
+    char *contents;
+    struct dirent_cs *next;
+    struct dirent_cs *prev;
+} dirent_ct;
+
+#define END_OF_DIRENT_LIST(LIST, DC) ((DC)->next == (LIST))
+#define START_OF_DIRENT_LIST(LIST, DC) ((DC) == (LIST))
+
+#define FOR_EACH_DIRENT_IN_LIST(LIST, VAR) \
+    for (VAR = LIST; VAR; VAR = (VAR->next == LIST) ? NULL : VAR->next)
+
+void dump_dirent_in_list (dirent_ct *list);
+void prepend_dirent_to_list (dirent_ct **list, dirent_ct *dc);
+void append_dirent_to_list (dirent_ct **list, dirent_ct *dc);
+void remove_dirent_from_list (dirent_ct **list, dirent_ct *dc);
+void sort_files (dirent_ct **list, int ascending);
+void free_files (dirent_ct **list);
+int move_file (dirent_ct *dc, const char *subdir);
+
+
+void
+dump_dirent_in_list (dirent_ct *list)
+{
```

Comment:
Please follow our coding style.
```
static void dump_dirent_in_list(dirent_ct *list) {
```

Reply:
I've applied this style to all functions in pstore.c